### PR TITLE
Fix D-Scanner linting issues

### PR DESCRIPTION
**Pull request by Vladiwostok**
This PR contains only code format changes.

There are a couple of linting issues that are now detected with the dmd-as-a-lib D-Scanner.

* UndocumentedDeclarationCheck => detects undocumented public declarations; the fix consists of changing the normal comment chars to DDoc comment chars and documenting the unit test;
* IfConstraintsIndentCheck => detects static ifs that are not on separate lines nor to they match the template's column indentation; the fix consists of moving the if conditions on separate lines and aligning them with the template declaration.

### CODE REVIEW

File: std/random.d [hunk 1]
```diff
@@ -2153,7 +2154,7 @@ if (isFloatingPoint!(CommonType!(T1, T2)) && isUniformRNG!UniformRandomNumberGen
 }
 
 // Implementation of uniform for integral types
-/+ Description of algorithm and suggestion of correctness:
+/** Description of algorithm and suggestion of correctness:
```
Comment (dkorpel) [inline std/random.d:2157]: This comment doesn't look intended for the documentation, and there's simple missing a `/// ditto`

File: std/algorithm/searching.d [hunk 2]
```diff
@@ -1425,8 +1424,7 @@ if (isInputRange!Range && !isInfinite!Range &&
 }
 
 private auto extremum(alias selector = "a < b", Range)(Range r)
-if (isInputRange!Range && !isInfinite!Range &&
-    !is(typeof(unaryFun!selector(ElementType!(Range).init))))
+if (isInputRange!Range && !isInfinite!Range && !is(typeof(unaryFun!selector(ElementType!(Range).init))))
```
Comment (RazvanN7) [inline std/algorithm/searching.d:1427]: This change looks rather restrictive. The constraint is indented properly, but it's split on 2 lines to make it easier to read. Are you sure this is what we want?

File: std/array.d [hunk 3]
```diff
@@ -2332,8 +2332,7 @@ if (isInputRange!RoR &&
 
 /// Ditto
 ElementEncodingType!(ElementType!RoR)[] join(RoR)(RoR ror)
-if (isInputRange!RoR &&
-    isInputRange!(Unqual!(ElementType!RoR)))
+if (isInputRange!RoR && isInputRange!(Unqual!(ElementType!RoR)))
```
Comment (RazvanN7) [inline std/array.d:2335]: ditto.

**Review by RazvanN7 — COMMENTED** [review]
As I understand the check should limit itself to highlighting instances of unaligned if constraints. We should leave it to the user to then split the constraints as he/she sees fit.

**Comment by Vladiwostok** [comment]
> As I understand the check should limit itself to highlighting instances of unaligned if constraints. We should leave it to the user to then split the constraints as he/she sees fit.

I see. I just saw those conditions split on multiple lines and thought it would be easier to read if they can fit in a single line. Will revert those kind of changes

**Comment by CyberShadow** [comment]
> UndocumentedDeclarationCheck => detects undocumented public declarations; the fix consists of changing the normal comment chars to DDoc comment chars and documenting the unit test;

Careful, there are many cases where a declaration is not intended for public use but must be `public` for technical reasons. I think a way to override this check would be more useful.

**Comment by Vladiwostok** [comment]
> > UndocumentedDeclarationCheck => detects undocumented public declarations; the fix consists of changing the normal comment chars to DDoc comment chars and documenting the unit test;
> 
> Careful, there are many cases where a declaration is not intended for public use but must be `public` for technical reasons. I think a way to override this check would be more useful.

I agree. We can create an issue in D-Scanner to implement `@nolint` exception for UndocumentedDeclarationCheck.

**Comment by thewilsonator** [comment]
```
generated/linux/debug/64/publictests/std_random.d(2287): Error: template instance `isForwardRange!(typeof(gen))` template `isForwardRange` is not defined
generated/linux/debug/64/publictests/std_random.d(2287):        while evaluating: `static assert(isForwardRange!(typeof(gen)))`
generated/linux/debug/64/publictests/std_random.d(2296): Error: undefined identifier `meta` in package `std`, perhaps add `static import std.meta;`
generated/linux/debug/64/publictests/std_random.d(2350): Error: undefined identifier `meta` in package `std`, perhaps add `static import std.meta;`
```

File: std/random.d [hunk 4]
```diff
@@ -2277,6 +2279,7 @@ if ((isIntegral!(CommonType!(T1, T2)) || isSomeChar!(CommonType!(T1, T2))) &&
     return cast(ResultType)(lower + offset);
 }
 
+///
 @safe unittest
 {
     import std.conv : to;
```
Comment (thewilsonator) [inline std/random.d:2285]: Add `import std.meta : AliasSeq;` here

**Comment by Vladiwostok** [comment]
> > > UndocumentedDeclarationCheck => detects undocumented public declarations; the fix consists of changing the normal comment chars to DDoc comment chars and documenting the unit test;
> > 
> > 
> > Careful, there are many cases where a declaration is not intended for public use but must be `public` for technical reasons. I think a way to override this check would be more useful.
> 
> I agree. We can create an issue in D-Scanner to implement `@nolint` exception for UndocumentedDeclarationCheck.

Created issue: https://github.com/Dlang-UPB/D-scanner/issues/165

File: std/path.d [hunk 5]
```diff
@@ -1446,9 +1446,8 @@ private auto _withDefaultExtension(R, C)(R path, C[] ext)
         of segments to assemble the path from.
     Returns: The assembled path.
 */
-immutable(ElementEncodingType!(ElementType!Range))[]
-    buildPath(Range)(scope Range segments)
-    if (isInputRange!Range && !isInfinite!Range && isSomeString!(ElementType!Range))
+immutable(ElementEncodingType!(ElementType!Range))[] buildPath(Range)(scope Range segments)
```
Comment (Vladiwostok) [inline std/path.d:1449]: This line change is necessary in order to make the style check pass, since the old D-Scanner considers this bad formatted code

**Review by LightBender — APPROVED** [review]

**Pull request merged**